tests/ops/api/v1/endpoints/test_dataset_endpoints.py

```diff
@@ -1232,6 +1233,152 @@ def test_patch_datasets_bulk_create(
+        )
+        assert response.status_code == 200
+        response_body = json.loads(response.text)
+        assert len(response_body["succeeded"]) == 1
+        assert len(response_body["failed"]) == 0
+
+        field_masking_override_dataset = response_body["succeeded"][0]
+        field_masking_override_config = DatasetConfig.get_by(
+            db=db, field="fides_key", value="field_masking_override_test_dataset"
+        )
+        assert field_masking_override_config is not None
+        field_masking_override_ctl_dataset = field_masking_override_config.ctl_dataset
+        assert field_masking_override_ctl_dataset is not None
+        assert (
+            field_masking_override_dataset["fides_key"]
+            == "field_masking_override_test_dataset"
+        )
+        assert (
+            field_masking_override_dataset["name"]
+            == "Field Masking Override Test Dataset"
+        )
+        assert (
+            "Example of a dataset containing masking strategy override at the field-level."
+            in field_masking_override_dataset["description"]
+        )
+        assert len(field_masking_override_dataset["collections"]) == 11
+        assert len(field_masking_override_ctl_dataset.collections) == 11
+
+        field_masking_override_config.delete(db)
+        field_masking_override_ctl_dataset.delete(db)
+
+    def test_patch_datasets_invalid_field_masking_strategy_override(
+        self,
+        datasets_url,
+        api_client: TestClient,
+        db: Session,
+        generate_auth_header,
+        connection_config,
+    ) -> None:
+        auth_header = generate_auth_header(scopes=[DATASET_CREATE_OR_UPDATE])
+        example_dataset = load_dataset(
+            "data/dataset/postgres_example_invalid_masking_strategy_override_dataset.yml"
+        )
+        response = api_client.patch(
+            datasets_url, headers=auth_header, json=example_dataset
+        )
+        assert response.status_code == 200
+        response_body = json.loads(response.text)
+        assert len(response_body["succeeded"]) == 0
+        assert len(response_body["failed"]) == 1
+
+        # Confirm that postgres dataset matches the values we provided
+        field_masking_override_dataset = response_body["failed"][0]
+        assert (
+            field_masking_override_dataset["data"]["fides_key"]
+            == "postgres_example_invalid_masking_strategy_override"
+        )
+        assert (
+            field_masking_override_dataset["message"]
+            == "Masking strategy 'hash' with required secrets not allowed as an override."
+        )
+        field_masking_override_config = DatasetConfig.get_by(
+            db=db,
+            field="fides_key",
+            value="postgres_example_invalid_masking_strategy_override",
+        )
+        assert field_masking_override_config is None
+
+    def test_patch_datasets_invalid_field_masking_strategy_override_update(
```

Comment:
I see tests for create and update, do we need one for upsert?

Reply:
Not really. The endpoint we are testing does uses an upsert function. So by testing creation and update we are esentially testing upsert. Tho i did create two more tests thanks to this comment. One for a succeded  patch update and one for a failed put.